package.go

```diff
@@ -16,6 +18,9 @@ type Package interface {
 	// The name of the Go package. This is guaranteed to be unique.
 	GoName() Name
 
+	// JavaName gets the java package as defined in the proto FileDescriptorProto.FileOptions.
+	JavaName() Name
+
```

Comment:
There are some other options for Java -- OuterClassName, MultiClass etc. They probably belong here -- WDYT ?